AISKU/src/ApplicationInsightsContainer.ts

```diff
@@ -1,8 +1,45 @@
-import { throwUnsupported } from "@nevware21/ts-utils";
+import { hasWindow, isUndefined, throwUnsupported } from "@nevware21/ts-utils";
 import { AppInsightsSku } from "./AISku";
 import { IApplicationInsights } from "./IApplicationInsights";
 import { Snippet } from "./Snippet";
 
+/**
+ * Detects if the current environment is a server-side rendering environment.
+ * This is used to prevent the SDK from initializing in environments like
+ * Angular SSR in Cloudflare Workers where certain operations are prohibited.
+ * @returns {boolean} True if the environment appears to be server-side rendering
+ */
+export function isServerSideRenderingEnvironment(): boolean {
```

Comment:
No!
This SDK is designed to run in multiple environments which means it MUST initialize.
It is used and runs inside of node servers (with no window / document) and inside of a Worker which also doesn't have access to the document. As well as other environments that don't provides these.